isoutils/isoutils.go

```diff
@@ -0,0 +1,517 @@
+		[]string{"cp", "-pr", tmpPaths[clr_img_efi] + "/.", tmpPaths[clr_efi]},
+	}
+
+	for _, i := range cmds {
+		err := cmd.RunAndLog(i...)
+		if err != nil {
+			prg.Failure()
+			return err
+		}
+	}
+
+	/* create dirs in new efi img */
+	if _, err := os.Stat(tmpPaths[clr_efi] + "/EFI/systemd"); os.IsNotExist(err) {
+		os.MkdirAll(tmpPaths[clr_efi]+"/EFI/systemd", os.ModePerm)
+	}
+
+	/* Modify loader/entries/Clear-linux-*, add initrd= line and remove ROOT= and rootwait from kernel command line options */
+	entriesGlob, err := filepath.Glob(tmpPaths[clr_efi] + "/loader/entries/Clear-linux-*")
+	if err != nil || len(entriesGlob) != 1 {
+		prg.Failure()
+		log.Error("Failed to modify efi entries file")
+		return err
+	}
+
+	input, err := ioutil.ReadFile(entriesGlob[0])
+	if err != nil {
+		prg.Failure()
+		log.Error("Failed to read EFI entries file")
+		return err
+	}
+
+	/* Replace current options line with initrd information */
+	lines := strings.Split(string(input), "\n")
+	for i, line := range lines {
+		if strings.Contains(line, "options") {
+			lines[i] = "initrd /EFI/BOOT/initrd.gz"
+		}
+	}
+
+	/* Pull kernel options from FS, add them to the buffer, write the file */
+	optionsGlob, err := filepath.Glob(tmpPaths[clr_rootfs] + "/usr/lib/kernel/cmdline*")
+	if err != nil || len(optionsGlob) > 1 { // Fail if there's >1 kernel(s)
+		prg.Failure()
+		log.Error("Failed to determine kernel boot params for initrd")
+		return err
+	}
+	optionsFile, err := ioutil.ReadFile(optionsGlob[0])
+	if err != nil {
+		prg.Failure()
+		log.Error("Cannot read kernel options file from rootfs")
+		return err
+	}
+	lines = append(lines, "options "+string(optionsFile))
+
+	err = ioutil.WriteFile(entriesGlob[0], []byte(strings.Join(lines, "\n")), 0644)
+	if err != nil {
+		prg.Failure()
+		log.Error("Failed to write kernel boot paramaters file")
+		return err
+	}
+
+	/* Copy all required files to efiboot.img and finally unmount efiboot.img */
+	cmds = [][]string{
+		[]string{"cp", tmpPaths[clr_cdroot] + "/images/initrd.gz", tmpPaths[clr_efi] + "/EFI/Boot/initrd.gz"},
+		[]string{"cp", tmpPaths[clr_img_efi] + "/EFI/BOOT/BOOTX64.EFI", tmpPaths[clr_efi] + "/EFI/systemd/systemd-bootx64.efi"},
+		[]string{"umount", tmpPaths[clr_efi]}, //must umount before packaging in ISO or files will not be writ yet TODO: replace with syscall
+	}
+	for _, i := range cmds {
+		err := cmd.RunAndLog(i...)
+		if err != nil {
+			prg.Failure()
+			return err
+		}
+	}
+
+	prg.Success()
+	return err
+}
+
+func mkLegacyBoot(templatePath string, isolinuxFilesPath string) error {
+	msg := "Setting up BIOS boot with isolinux"
+	prg := progress.NewLoop(msg)
+	log.Info(msg)
+
+	type BootConf struct {
+		Options string
+	}
+	bc := BootConf{}
+
+	/* Find kernel path so we can copy the kernel later */
+	kernelGlob, err := filepath.Glob(tmpPaths[clr_rootfs] + "/lib/kernel/org.clearlinux.native.*")
```

Comment:
We will need to update this somewhat because we might not always have the native kernel; could be kvm or lts or ...:
```
# ls /lib/kernel/org.clearlinux.*
/lib/kernel/org.clearlinux.kvm.5.0.3-315
/lib/kernel/org.clearlinux.lts2018.4.19.30-15
/lib/kernel/org.clearlinux.native.5.0.3-721
```

Reply:
Done, same as previous - works with any kernel names now (as long as there's not an extra `.` in them...)